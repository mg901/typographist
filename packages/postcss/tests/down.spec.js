const run = require('./run');

describe('@down', () => {
  test('with breakpoints name', async () => {
    const source = `
            .test {
              @down (desktop) {
                background-color: gold;
                }
            }`;

    expect(await run(source)).toMatchSnapshot();
  });

  test('with pixels', async () => {
    const source = `
            .test {
              @down (1000px) {
                background-color: gold;
                }
            }`;

    expect(await run(source)).toMatchSnapshot();
  });
  test('with em', async () => {
    const source = `
            .test {
              @down (40em) {
                background-color: gold;
                }
            }`;

    expect(await run(source)).toMatchSnapshot();
  });

  test('with orientation `landscape`', async () => {
    const source = `
              .test {
                @down (desktop):landscape {
                  background-color: gold;
                  }
              }`;

    expect(await run(source)).toMatchSnapshot();
  });

  test('with orientation `portrait`', async () => {
    const source = `
              .test {
                @down (desktop):portrait {
                  background-color: gold;
                  }
              }`;

    expect(await run(source)).toMatchSnapshot();
  });
});
