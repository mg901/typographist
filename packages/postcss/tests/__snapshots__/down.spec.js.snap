// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@down with breakpoints name 1`] = `
@media (max-width: 74.99875em) {
  .test {
    background-color: gold;
  }
}
`;

exports[`@down with em 1`] = `
@media (max-width: 39.99875em) {
  .test {
    background-color: gold;
  }
}
`;

exports[`@down with orientation \`landscape\` 1`] = `
@media (max-width: 74.99875em) and (orientation: landscape) {
  .test {
    background-color: gold;
  }
}
`;

exports[`@down with orientation \`portrait\` 1`] = `
@media (max-width: 74.99875em) and (orientation: portrait) {
  .test {
    background-color: gold;
  }
}
`;

exports[`@down with pixels 1`] = `
@media (max-width: 62.49875em) {
  .test {
    background-color: gold;
  }
}
`;
